src/shared/mount-util.c

```diff
@@ -765,7 +765,9 @@ int mount_option_mangle(
                 }
 
                 /* If 'word' is not a mount flag, then store it in '*ret_remaining_options'. */
-                if (!ent->name && !strextend_with_separator(&ret, ",", word))
+                if (!ent->name &&
+                    !streq(word, "x-systemd.extension-release.nonstrict") &&
```

Comment:
why drop this here? it shouldn't hurt if we pass this along? and might be good for debugging to let it propagate this on. libmount should eat it up once passing this to the kernel, no?

Reply:
nope, the mount syscall fails if passed through

Reply:
oh, hmm, this is not going through libmount here. hmm. ok.

but then maybe just ignore everything that starts with `x-` and `X-` maybe. To match at least one tiny subset of what libmount wuld do.

Reply:
I'm a bit wary of introducing regressions, so I'll ignore x-systemd* rather than x*, we shouldn't be using anything else in our codebase anyway

Reply:
it's official libmount API, and documented. It should be safe to filter `x-*.`/`X-*.`, because that's documented clearly as userspace mount options and is filtered by libmount too.

Reply:
ok, updated